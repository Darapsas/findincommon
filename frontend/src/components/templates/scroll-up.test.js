import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import ScrollUp from './scroll-up'

global.scrollTo = jest.fn()

describe('<ScrollUp />', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<ScrollUp />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('contains a button', () => {
    const wrapper = shallow(<ScrollUp />)
    expect(wrapper.find('button').exists()).toBeTruthy()
  })
  it('button scrolls the page to the top position', () => {
    const wrapper = shallow(<ScrollUp />)
    const scrollUpButton = wrapper.find('button')
    scrollUpButton.simulate('click')
    expect(global.scrollTo).toHaveBeenCalledWith(0, 0)
  })
})
