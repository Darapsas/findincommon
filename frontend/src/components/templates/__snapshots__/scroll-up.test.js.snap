// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollUp /> matches the snapshot 1`] = `
<button
  className="btn btn-sm btn-outline-secondary"
  onClick={[Function]}
  type="button"
>
  Back to top
</button>
`;
